_posts/2024-XX-XX-quarkus-kiota.adoc

```diff
@@ -0,0 +1,160 @@
+
+To pave the way for better tooling, new standards are being defined in the computer industry. For building HTTP-based APIs, link:https://www.openapis.org[OpenAPI] stands out as a widely adopted option.
+In the context of this article we are going to assume you already have an OpenAPI description of the service.
+
+=== A Story of Open Source Collaboration
+
+==== Microsoft Kiota
+
+Recognizing the challenges faced by the industry in generating comprehensive and efficient SDKs for diverse APIs, Microsoft took the lead. In the context of link:https://learn.microsoft.com/en-us/graph/overview[Microsoft Graph], they introduced link:https://github.com/microsoft/kiota[Kiota], a CLI to streamline and automate the process of creating Software Development Kits (SDKs) for HTTP APIs.
+
+Kiota, born and raised in the open source, fosters collaboration and is ground-up designed with link:https://learn.microsoft.com/en-gb/openapi/kiota/design[modularity] in mind. The Kiota project's community is extra welcoming, and it has demonstrated to be easy to open an issue not only for bugs but also to improve over the design decisions.
+
+Nowadays, Kiota demonstrates its value daily with the automated generation of SDKs for an incredible number of Microsoft services. More big industries, like link:https://github.blog/2024-01-03-our-move-to-generated-sdks/[GitHub], are quickly following the lead.
+
+==== Red Hat
+
+Deeply understanding the technological problem and accepting the fact that the available alternatives are sub-optimal, especially for consistency across languages and frameworks, Red Hat joined the effort on Kiota. The collaboration led to important milestones, directly blossomed from this collaboration:
+
+- To make the usage of the generated code almost indistinguishable from a human-crafted product, we thoroughly implemented language-specific mangling of names.
+- To ease the fruition of the Kiota CLI from standard Java projects, we implemented a link:https://github.com/kiota-community/kiota-java-extra?tab=readme-ov-file#maven-plugin[Maven plugin].
+- To make frictionless the usage of Kiota on different and opinionated technological stacks, we ship alternative core libraries to let the user easily swap out from Microsoft's defaults (OkHttp + Gson): link:https://github.com/kiota-community/kiota-java-extra?tab=readme-ov-file#serialization-jackson[Jackson Serde], link:https://github.com/kiota-community/kiota-java-extra?tab=readme-ov-file#http-vertx[Vert.X Http], link:https://github.com/kiota-community/kiota-java-extra?tab=readme-ov-file#http-jdk[JDK Http].
```

Comment:
```suggestion
- To make frictionless the usage of Kiota on Quarkus, we ship alternative core libraries to let the user easily swap out from Microsoft's defaults (OkHttp + Gson): link:https://github.com/kiota-community/kiota-java-extra?tab=readme-ov-file#serialization-jackson[Jackson Serde], link:https://github.com/kiota-community/kiota-java-extra?tab=readme-ov-file#http-vertx[Vert.X Http], link:https://github.com/kiota-community/kiota-java-extra?tab=readme-ov-file#http-jdk[JDK Http].
```

...is there another stack than Quarkus where those choices makes sense?

Reply:
It depends on the stack people are using, Jackson is widely used instead of Gson AFAIK, but happy to revisit this.